packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2638,29 +2641,53 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this.logger.sendErrorEvent({ eventName: "SubmitOpInReadonly" });
         }
 
+        const message: BatchMessage = {
+            contents: serializedContent,
+            deserializedContent,
+            metadata,
+            localOpMetadata,
+            referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
+        };
+
         try {
-            this.batchManager.push({
-                contents: serializedContent,
-                deserializedContent,
-                metadata,
-                localOpMetadata,
-                referenceSequenceNumber: this.deltaManager.lastSequenceNumber,
-            });
+            // If this is attach message for new data store, and we are in a batch, send this op out of order
+            // Is it safe:
+            //    Yes, this should be safe reordering. Newly created data stores are not visible through API surface.
```

Comment:
I am wondering if it can affect scenarios where data stores are requested (via request or resolveHandle APIs) with the "wait" header set to true (which is also the default is the header is not specified).
For example, consider the following scenario which is not a common scenario but can still happen:
- Client1 requests for a data store and is waiting for it to show up so it can perform operations on it.
- Client2 creates this data store and sends attach op before other ops that are queued but are not sent yet.
- Client1 loads the data store, completes the pending request and sends ops for the data store. These ops might get sequenced before the ops from Client2 which were supposed to be before any activity for this new data store.

I am not sure if this is real use case or not, but it seems like this would end up changing the order of how things appear from a client's perspective.

Reply:
Yes, if you can guess guid :) The whole safety premise is based on assumption that user provided (root) name is visible only after attach op. As long as it stays after attach op, we are not changing visibility of this data store - it is invisible to the system. Maybe other way to say it - as long as we do not provide "enumerate" API for data stores, we should be good.

Reply:
Perfect, that makes sense! :-)